uiTests/src/test/resources/project.rhino.js

```diff
@@ -0,0 +1,68 @@
+
+importClass(com.intellij.ide.DataManager);
+importClass(com.intellij.openapi.actionSystem.ActionManager);
+importClass(com.intellij.openapi.actionSystem.AnActionEvent);
+importClass(com.intellij.openapi.actionSystem.Presentation);
+importClass(com.intellij.openapi.wm.ToolWindowId);
+importClass(com.intellij.openapi.wm.ToolWindowManager);
+importClass(com.intellij.ui.GuiUtils);
+
+// Do not run any of the methods in EDT.
+function Project(underlyingProject) {
+  this.openTab = function () {
+    const toolWindowManager = ToolWindowManager.getInstance(underlyingProject);
+    let toolWindow;
+    do {
+      toolWindow = toolWindowManager.getToolWindow(ToolWindowId.VCS);
+      sleep();
+    } while (toolWindow == null);
+
+    // The test is (obviously) not run on UI thread,
+    // so `runOrInvokeAndWait` really means `enqueue onto UI thread and wait until complete`.
```

Comment:
`onto the UI thread` :question: :gammar-lover:

Reply:
Fixed everywhere in the project